test/SvgGenerationTest.hx

```diff
@@ -0,0 +1,280 @@
+    private function diffPixelsRgba(expectedPixel:Array<Int>, actualPixel:Array<Int>):Float
+    {
+        // Special case: if expected or actual are [0, 0, 0, 0] but the other isn't, return 100%
+        // Usually, all four components are zero. But we don't check, because it makes the if
+        // statement very, very tedious to read.
+        var expectedIsZero = expectedPixel[3] == 0;
+        var actualIsZero = actualPixel[3] == 0;
+        if ((expectedIsZero || actualIsZero) && !(expectedIsZero && actualIsZero))
+        {
+            return 1;
+        }
+        var redDiff:Float = Math.abs(expectedPixel[0] - actualPixel[0]) / 255.0;
+        var greenDiff:Float = Math.abs(expectedPixel[1] - actualPixel[1]) / 255.0;
+        var blueDiff:Float = Math.abs(expectedPixel[2] - actualPixel[2]) / 255.0;
+        var alphaDiff:Float = Math.abs(expectedPixel[3] - actualPixel[3]) / 255.0;
+        
+        // Average of RGBA diffs
+        var percentDiff = (redDiff + greenDiff + blueDiff + alphaDiff) / 4;
+        return percentDiff;
+    }
+
+	// Creates the HTML report
+	private function createHtmlReport(results:GenerationResults)
+	{
+		var html:String = '<html><head>
+		<title>${results.failedTests.length} failures | SVG Generation Tests</title>
+		</head><body style="background-color: #eee">';
+
+		// TODO: beautify HTML.
+		var total = results.failedTests.length + results.passedTests.length;
+		// Failures first, because we care about fixing those
+		html += createTableFor(results.failedTests, "Failures");
+		html += createTableFor(results.passedTests, "Successes");
+
+		html = '${html}</body></html>';
+
+		if (sys.FileSystem.exists(RESULTS_HTML_FILE)) {
+			sys.FileSystem.deleteFile(RESULTS_HTML_FILE);
+		}
+		sys.io.File.saveContent(RESULTS_HTML_FILE, html);
+	}
+
+	private function createTableFor(tests:Array<SvgTest>, header:String) : String
+	{
+		var html:String = '<h1>${tests.length} ${header}</h1>';
+		html += "<table><tr>
+			<th>Expected (PNG)</th>
+			<th>Actual (PNG)</th>
+            <th>Diff Image</th>
+            <th>Average Pixel Diff %</th>";
+            
+
+		for (test in tests) {
+			var pngFile = test.fileName.replace('.svg', '.png');
+			var diffFile = test.fileName.replace('.svg', '-diff.png');
+			html += '<tr>
+				<td><img src="${IMAGES_PATH}/${pngFile}" /></td>
+				<td><img src="${GENERATED_IMAGES_PATH}/${pngFile}" /></td>
+				<td><img src="${GENERATED_IMAGES_PATH}/${diffFile}" /></td>                
+                <td>${test.diffPercentage * 100}%</td>
+			</tr>';
+		}
+		html += "</table>";
+		return html;
+	}
+    
+    // Given a pixel (0xAARRGGBB), return an array [RR, GG, BB, AA]
+    // Components are integer values from 0..255
+    private function getComponents(pixel:Int):Array<Int>
+    {
+        // No difference (empty pixel)? Skip calculations.
+        if (pixel <= 0)
+        {
+            return [0, 0, 0, 0];
+        }
+        
+        var blue:Int = 0xFFBB8844 & 0xFF; // BB
```

Comment:
Yes, you're right. I'll fix it. (I added this initially for testing and forgot to undo it.)